# disallow HTML comments. (ex. :ng: `<!-- comment -->`) (lodash-template/no-html-comments)

- :gear: This rule is included in all of `"plugin:lodash-template/best-practices"`, `"plugin:lodash-template/recommended"` and `"plugin:lodash-template/all"`.

## Rule Details

This rule reports HTML comments.

:-1: Examples of **incorrect** code for this rule:

```html
<!-- comment -->
```
