# disallow HTML comments. (ex. :-1: `<!-- comment -->`) (lodash-template/no-html-comments)

- :gear: This rule is included in `"plugin:lodash-template/best-practices"` and `"plugin:lodash-template/recommended"`.

## Rule Details

This rule reports HTML comments.

:-1: Examples of **incorrect** code for this rule:

```html
<!-- comment -->
```
