---
sidebarDepth: 0
---

# All Rules

<!-- This file is automatically generated in tools/update-docs-rules-index.js, do not change! -->

## Base Rules (Enabling Correct ESLint Parsing)

Enable this plugin using with:

```json
{
  "extends": "plugin:lodash-template/base"
}
```

| Rule ID | Description |    |
|:--------|:------------|:---|
| [lodash-template/no-script-parsing-error](./no-script-parsing-error.md) | disallow parsing errors in template |  |

## Best Practices (Improve Development Experience)

Enforce all the rules in this category with:

```json
{
  "extends": "plugin:lodash-template/best-practices"
}
```

| Rule ID | Description |    |
|:--------|:------------|:---|
| [lodash-template/no-empty-template-tag](./no-empty-template-tag.md) | disallow empty micro-template tag. (ex. :ng: `<% %>`) |  |
| [lodash-template/no-invalid-template-interpolation](./no-invalid-template-interpolation.md) | disallow other than expression in micro-template interpolation. (ex. :ng: `<%= if (test) { %>`) |  |
| [lodash-template/no-semi-in-template-interpolation](./no-semi-in-template-interpolation.md) | disallow the semicolon at the end of expression in micro template interpolation.(ex. :ok: `<%= text %>` :ng: `<%= text; %>`) | :wrench: |

## Recommended (Improve Readability)

Enforce all the rules in this category and all the rules in `Best Practices` categories with:

```json
{
  "extends": "plugin:lodash-template/recommended"
}
```

| Rule ID | Description |    |
|:--------|:------------|:---|
| [lodash-template/no-irregular-whitespace](./no-irregular-whitespace.md) | disallow irregular whitespace outside the template tags. | :wrench: |
| [lodash-template/no-multi-spaces-in-scriptlet](./no-multi-spaces-in-scriptlet.md) | disallow multiple spaces in scriptlet. (ex. :ng: `<% if···(test)···{ %>`) | :wrench: |
| [lodash-template/scriptlet-indent](./scriptlet-indent.md) | enforce consistent indentation to scriptlet in micro-template tag. | :wrench: |
| [lodash-template/template-tag-spacing](./template-tag-spacing.md) | enforce unified spacing in micro-template tag. (ex. :ok: `<%= prop %>`, :ng: `<%=prop%>`) | :wrench: |

## Recommended with HTML template (Improve Readability with HTML template)

Enforce all the rules in this category and all the rules in `Best Practices`/`Recommended` categories with:

```json
{
  "extends": "plugin:lodash-template/recommended-with-html"
}
```

| Rule ID | Description |    |
|:--------|:------------|:---|
| [lodash-template/attribute-name-casing](./attribute-name-casing.md) | enforce HTML attribute name casing. (ex. :ok: `<div foo-bar>` :ng: `<div fooBar>` `<div FOO-BAR>`) | :wrench: |
| [lodash-template/attribute-value-quote](./attribute-value-quote.md) | enforce quotes style of HTML attributes. (ex. :ok: `<div class="abc">` :ng: `<div class='abc'>` `<div class=abc>`) | :wrench: |
| [lodash-template/element-name-casing](./element-name-casing.md) | enforce HTML element name casing. (ex. :ok: `<xxx-element>` :ng: `<xxxElement>` `<DIV>`) | :wrench: |
| [lodash-template/html-closing-bracket-newline](./html-closing-bracket-newline.md) | require or disallow a line break before tag's closing brackets | :wrench: |
| [lodash-template/html-closing-bracket-spacing](./html-closing-bracket-spacing.md) | require or disallow a space before tag's closing brackets. (ex. :ok: `<input>` `<input·/>` :ng: `<input·>` `<input/>`) | :wrench: |
| [lodash-template/html-comment-content-newline](./html-comment-content-newline.md) | require or disallow a line break before and after HTML comment contents | :wrench: |
| [lodash-template/html-comment-spacing](./html-comment-spacing.md) | enforce unified spacing in HTML comment. (ex. :ok: `<!-- comment -->`, :ng: `<!--comment-->`) | :wrench: |
| [lodash-template/html-content-newline](./html-content-newline.md) | require or disallow a line break before and after HTML contents | :wrench: |
| [lodash-template/html-indent](./html-indent.md) | enforce consistent HTML indentation. | :wrench: |
| [lodash-template/max-attributes-per-line](./max-attributes-per-line.md) | enforce the maximum number of HTML attributes per line | :wrench: |
| [lodash-template/no-duplicate-attributes](./no-duplicate-attributes.md) | disallow duplication of HTML attributes. (ex. :ng: `<div foo foo>`) |  |
| [lodash-template/no-html-comments](./no-html-comments.md) | disallow HTML comments. (ex. :ng: `<!-- comment -->`) |  |
| [lodash-template/no-multi-spaces-in-html-tag](./no-multi-spaces-in-html-tag.md) | disallow multiple spaces in HTML tags. (ex. :ng: `<input···type="text">`) | :wrench: |
| [lodash-template/no-space-attribute-equal-sign](./no-space-attribute-equal-sign.md) | disallow spacing around equal signs in attribute. (ex. :ok: `<div class="item">` :ng: `<div class = "item">`) | :wrench: |
| [lodash-template/no-warning-html-comments](./no-warning-html-comments.md) | disallow specified warning terms in HTML comments. (ex. :ng: `<!-- TODO:task -->`) |  |

## Uncategorized

No preset enables the rules in this category.
Please enable each rule if you want.

For example:

```json
{
  "rules": {
    "undefined": "error"
  }
}
```

| Rule ID | Description |    |
|:--------|:------------|:---|
| [lodash-template/no-template-tag-in-start-tag](./no-template-tag-in-start-tag.md) | disallow template tag in start tag outside attribute values. (ex. :ng: `<input <%= 'disabled' %> >`) |  |
| [lodash-template/prefer-escape-template-interpolations](./prefer-escape-template-interpolations.md) | prefer escape micro-template interpolations. (ex. :ok: `<%- ... %>`, :ng: `<%= ... %>`) |  |

## Deprecated

- :warning: We're going to remove deprecated rules in the next major release. Please migrate to successor/new rules.
- :innocent: We don't fix bugs which are in deprecated rules since we don't have enough resources.

| Rule ID | Replaced by |
|:--------|:------------|
| [lodash-template/no-multi-spaces-in-script](./no-multi-spaces-in-script.md) | [lodash-template/no-multi-spaces-in-scriptlet](./no-multi-spaces-in-scriptlet.md) |
| [lodash-template/plugin-option](./plugin-option.md) | (no replacement) |
| [lodash-template/script-indent](./script-indent.md) | [lodash-template/scriptlet-indent](./scriptlet-indent.md) |
